tap_chromedata/streams.py

```diff
@@ -7,518 +8,204 @@
 
-# TODO: Delete this is if not using json files for schema definition
-#SCHEMAS_DIR = Path(__file__).parent / Path("./schemas")
-# TODO: - Override `UsersStream` and `GroupsStream` with your own stream definition.
-#       - Copy-paste as many times as needed to create multiple stream types.
 
+def camel_to_snake(name):
+    """Convert camelCase words to snake_case"""
+    name = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', name).lower()
+    
 
-class QuickDataStream(Stream):
-    """Define custom stream."""
-    name = "QuickData"
-    primary_keys = ["autobuilder_style_id"]
-    replication_key = None
-    # Optionally, you may also use `schema_filepath` in place of `schema`:
-    # schema_filepath = SCHEMAS_DIR / "users.json"
-    schema = th.PropertiesList(
-        th.Property("model_year", th.IntegerType),
-        th.Property("division_name", th.StringType),
-        th.Property("model_name", th.StringType),
-        th.Property("hist_style_id", th.IntegerType),
-        th.Property("style_name", th.StringType),
-        th.Property("style_name_wo_trim", th.StringType),
-        th.Property("trim", th.StringType),
-        th.Property("full_style_code", th.StringType),
-        th.Property("style_sequence", th.IntegerType),
-        th.Property("msrp", th.NumberType),
-        th.Property("invoice", th.NumberType),
-        th.Property("destination", th.NumberType),
-        th.Property("model_effective_date", th.StringType),
-        th.Property("model_comment", th.StringType),
-        th.Property("manufacturer_name", th.StringType),
-        th.Property("manufacturer_id", th.IntegerType),
-        th.Property("division_id", th.IntegerType),
-        th.Property("hist_model_id", th.IntegerType),
-        th.Property("market_class", th.StringType),
-        th.Property("market_class_id", th.IntegerType),
-        th.Property("subdivision_name", th.StringType),
-        th.Property("subdivision_id", th.IntegerType),
-        th.Property("style_id", th.IntegerType),
-        th.Property("autobuilder_style_id", th.StringType)
-    ).to_dict()
+class ChromeDataStream(Stream):
+    """Base stream class with config parameter getters, tranversers to ACES Mapping data and data cleaning"""
+    flo=""
+    dirname=""
+    mainfilename=""
     @property
     def url_base(self) -> str:
         """Return the API URL root, configurable via tap settings."""
-        #print(self.config["FTP_URL"],self.config["FTP_USER"],self.config["FTP_PASS"])
-        return self.config["FTP_URL"]
+        return self.config["tap_chromedata_ftp_url"]
     @property
     def url_user(self) -> str:
-        """Return the API URL root, configurable via tap settings."""
-        #print(self.config["FTP_URL"],self.config["FTP_USER"],self.config["FTP_PASS"])
-        return self.config["FTP_USER"]
+        """Return the API URL user, configurable via tap settings."""
+        return self.config["tap_chromedata_ftp_user"]
```

Comment:
you don't need to namespace these, as all config options are for chrome data, so it's redundant.


for example:

A tap declares the configuration options it takes, and these are usually simple, e.g. `config.json` `{"ftp_user":"someone@example.com","ftp_pass":"secretpassword"}`.

In `meltano.yml`, the plugin declares in `settings` what that taps properties are. This tells Meltano _how_ to generate `config.json`:

```yaml
extractors:
- name: tap-chromedata
  settings:
    - name: ftp_user
    - name: ftp_pass
      kind: password
```

Then, also in `meltano.yml`, the configuration for _our instance_ of the tap goes in `config`:

```yaml
extractors:
- name: tap-chromedata
  # ...
  config:
    ftp_user: fixduser
    ftp_password: $TAP_CHROMEDATA_FTP_PASSWORD
```

There are two things going on here with respect to environment variables:

1. Because Meltano knows all of the settings, it allows using environment variables for config options, even to override the values set in `config`. Any setting can be changed with an environment variable like `inflection.underscore(upper(f"{tap_name}_{setting_name}"))`, in this case `TAP_CHROMEDATA_FTP_USER` and `TAP_CHROMEDATA_FTP_PASSWORD`. Running meltano with `TAP_CHROMEDATA_FTP_USER=someotheruser` would override the `fixduser` value. Meltano _has_ to namespace these, since there could be conflicts between plugins (think two plugins with an `api_key` option for example) and environment variables are global (unlike values set explictly in `config`).

2. I set `ftp_password: $TAP_CHROMEDATA_FTP_PASSWORD` in that example to an environment variable so I didn't have to hard code a secret in the code, but I didn't have to specify anything. Because of 1), we could leave the `ftp_password` out of the `config` section altogether and Meltano would still know to look for `TAP_CHROMEDATA_FTP_PASSWORD` (because of the `TAP_NAME_SETTING_NAME` convention).

It would also be legal to change the name of environment variable used:

```yaml
extractors:
- name: tap-chromedata
  # ...
  config:
    ftp_user: fixduser
    ftp_password: $CUSTOM_FTP_PASSWORD
```

Here, this would instruct Meltano to look for the password in _either_ `CUSTOM_FTP_PASSWORD` or `TAP_CHROMEDATA_FTP_PASSWORD`. Still, unless there's a good reason to use a different environment variable name, using `TAP_CHROMEDATA_` keeps things clear and organized. And explicitly setting the value to `ftp_password: $TAP_CHROMEDATA_FTP_PASSWORD` also serves as documentation to other developers.

Does that make sense?